duckduckgo/Android
# Improve the way we choose the favicon placeholder letter for autofill screens

Task/Issue URL: https://app.asana.com/0/0/1203994780181874/f

### Description
Updates the favicon placeholder used when we can't get a real favicon in the autofill management screens. 

### Steps to test this PR

Note, favicons won't show immediately for a manually added site, so you will typically see the placeholder in these scenarios (they'll be cached after next visiting that site so it will show thereafter).

For each of the following; verify the favicon placeholder is correct in **both** management list screen and when clicking into the details for that credential.

- [x] Add a login with only a text-based title (e.g. "example); verify the first letter of the title is used "e" for placeholder
- [x] Add a login with only a numerical title (e.g., 192); verify the `#` placeholder is used
- [x] Add a login with only a simple domain (e.g., `example.com`; verify `E` used as placeholder
- [x] Add a login with only a subdomain (e.g., `foo.example.com`; verify first letter of E-TLD+1 used as placeholder (e.g., "E")
- [x] Add a login with both title and url; verify the first letter of the title is used
